### PR TITLE
feat: added prev_owner to response log

### LINKED ISSUE #11: can add prev_owner/from attributes at transfer_nft and send_nft?
When owner approve the token and contract execute transfer, we can't know where is the token came from by the logs.

This is really hard to track token transfer. I made an example on terra.

1. User A make approve to transfer the cw721 token to contract B  
https://finder.terra.money/columbus-5/tx/14D4251F41053E7572A740D7FDB1760A2D3FD593870B0D70FB1EDBCA7121F211

2. User C execute the contract B to send it to B
https://finder.terra.money/columbus-5/tx/AEA9F3EEF1AD4A29644355777BD4157422E55E50130C2D330BC6EC0F4E63F51F

The token transferred to B from A. But there is no information about that at tx of 2.

Because of this Cw20 have transfer_from and send_from.

It is possible to make the attribute that can show the previous owner?

### PR DESCRIPTION
**Pull request by edison0xyz**
Resolves https://github.com/CosmWasm/cw-nfts/issues/11

Useful feature and have added it in.

### CODE REVIEW

File: contracts/cw721-base/src/contract_tests.rs [hunk 1]
```diff
@@ -194,6 +194,7 @@ fn transferring_nft() {
         Response::new()
             .add_attribute("action", "transfer_nft")
             .add_attribute("sender", "venus")
```
Comment (JakeHartnell) [inline contracts/cw721-base/src/contract_tests.rs:196]: Language is a bit confusing her now... we have `sender`, `ricpient`, and `from`...

Comment (JakeHartnell) [inline contracts/cw721-base/src/contract_tests.rs:196]: Almost like @larry0x's comment of using `from` and `to` like the Ethereum transfer events.

It is important to maintain sender, as that may be different than the owner, even though there is a little duplication there.

Comment (JakeHartnell) [reply contracts/cw721-base/src/contract_tests.rs:196]: Also, whatever we decide should be consistent with `SendNft`.

**Review by JakeHartnell — CHANGES_REQUESTED** [review]
`TransferNFT` needs to be consistent with `SendNFT`.

**Comment by JakeHartnell** [comment]
What do folks think about using `to` over `recipient`? Mirroring Ethereum: https://ethereum.org/en/developers/docs/standards/tokens/erc-721/#events

**Comment by the-frey** [comment]
FWIW I'm in favour of sticking close to the ETH standards where possible. `from` and `to` are probably the way to go in that case IIUC.

**Comment by iLoveBusinessDevelopment** [comment]
I think we should continue with `recipient`. CW20 is already widely used at this point and uses `recipient`.

**Comment by the-frey** [comment]
Hmmm, I suppose I'm looking at the POV of devs who are coming from ETH/ERC721 land. Although can see the argument for CW20 style too 🤷

**Comment by edison0xyz** [comment]
I think that we might be better to stick with the current CW standards.
Though i see the point of standardising with the Ethereum contracts, but there should be consistency amongst the CosmWasm contracts.

Changing everything to Ethereum sounds like a lot of work in code-change and communication with exisiting protocols. So perhaps let's avoid that.

**Review by orkunkl — APPROVED** [review]
LGTM

**Comment by ALPAC-4** [comment]
And I think we need to add owner at mint response log

https://github.com/CosmWasm/cw-nfts/blob/2712c05fbc365a44809c904f8508410c4f70383e/contracts/cw721-base/src/execute.rs#L111

I think owner is more important information than minter. Minter is always same for same contract

**Comment by shanev** [comment]
This issue is a bit stale. Okay to close?

**Comment by JakeHartnell** [comment]
> This issue is a bit stale. Okay to close?

Yeah.

**Comment by peara** [comment]
@JakeHartnell @larry0x  @shanev Please reconsider this PR. This is very useful for indexers and apps and for the consistency of cw721 as well.
I think adding `owner` as already implemented is good enough. The only issue remaining is adding the same to `send_nft` and `mint`, is that right?

**Pull request closed without merging**